src/core/speedy-media-source.js

```diff
@@ -710,3 +712,102 @@ class SpeedyBitmapMediaSource extends SpeedyMediaSource
         return new SpeedyBitmapMediaSource(PRIVATE_TOKEN)._load(bitmap);
     }
 }
+
+/**
+ * ImageData media source:
+ * a wrapper around ImageData
+ */
+class SpeedyImageDataMediaSource extends SpeedyMediaSource {
+    /**
+     * @private Constructor
+     * @param {symbol} token
+     */
+    constructor(token) {
+        super(token);
+        this._data = null;
+    }
+
+    /**
+     * The underlying wrapped object
+     * @returns {ImageData}
+     */
+    get data() {
+        return this._data;
+    }
+
+    /**
+     * The type of the underlying media source
+     * @returns {MediaType}
+     */
+    get type() {
+        return MediaType.ImageData;
+    }
+
+    /**
+     * Media width, in pixels
+     * @returns {number}
+     */
+    get width() {
+        return this._data ? this._data.width : 0;
+    }
+
+    /**
+     * Media height, in pixels
+     * @returns {number}
+     */
+    get height() {
+        return this._data ? this._data.height : 0;
+    }
+
+    /**
+     * Clone this media source
+     * @returns {SpeedyPromise<SpeedyMediaSource>}
+     */
+    clone() {
+        if (this._data == null)
+            throw new IllegalOperationError(`Media not loaded`);
+
+        const imageDataCopy = new ImageData(
+            new Uint8ClampedArray(this._data.data),
+            this._data.width,
+            this._data.height
+        )
+
+        return SpeedyImageDataMediaSource.load(imageDataCopy);
+    }
+
+    /**
+     * Release resources associated with this object
+     * @returns {null}
+     */
+    release() {
+        if (this._data != null)
+            this._data.close();
+
```

Comment:
Unlike an `ImageBitmap`, an `ImageData` does not have a `close()` method. We should just remove these lines.

Reply:
Ok i will remove It.

Reply:
So it should be:

```js
 release() {
        return super.release();
    }
```
does it correct?

Reply:
It is correct, but since this method does nothing, we can just remove it altogether. As you can see, most other media sources don't implement it.

Reply:
Ouch! You are right i will  when  i came back from holiday.🙂